src/ecies.cpp

```diff
@@ -73,7 +73,7 @@ class ECIESCtxBuilder::Impl
             return std::make_unique<mococrw::HMAC>(openssl::DigestTypes::SHA512, key);
         });
 
-        setMacKeySize(512 / 8);
+        setMacKeySize(Hash::getDigestSize(openssl::DigestTypes::SHA512));
```

Comment:
Commit message:

`s/the unit the testWithSalt/the unit test testWithSalt/`

`s/in a late comment/in a later comment/` or `s/in a late comment/in a comment later/`